File: scripts/scala/scala_2.10_da_prog_funcional_as_novas_features_java/src/com/github/diegopacheco/sandbox/java/caseclasses/CaseClassMain.java
package com.github.diegopacheco.sandbox.java.caseclasses;

public class CaseClassMain {
	
	static class Pessoa{
		private String  nome;
		private String  endereco = "centro";
		private Integer telefone = 0;
		
		public Pessoa(String name) {
			this.nome = name;
		}

		public Pessoa(String name, String endereco, Integer telefone) {
			this.nome = name;
			this.endereco = endereco;
			this.telefone = telefone;
		}

		public String getName() {
			return nome;
		}
		public void setName(String name) {
			this.nome = name;
		}

		public String getEndereco() {
			return endereco;
		}
		public void setEndereco(String endereco) {
			this.endereco = endereco;
		}

		public Integer getTelefone() {
			return telefone;
		}
		public void setTelefone(Integer telefone) {
			this.telefone = telefone;
		}
		
	}
	
	public static void main(String[] args) {
		System.out.println( new Pessoa("Diego").getName() );
	}
	
}

File: scripts/scala/scala_2.10_da_prog_funcional_as_novas_features_scala/src/com/github/diegopacheco/sandbox/scala/caseclasses/CaseClassMain.scala
package com.github.diegopacheco.sandbox.scala.caseclasses

import scala.beans.BeanProperty

case class Pessoa(@BeanProperty var nome:String, 
				  @BeanProperty var Endereco:String = "centro", 
				  @BeanProperty var telefone:Int = 0)

object CaseClassMain extends App {
	println( Pessoa(nome = "Diego").getNome )
}

